api/git/git.go

```diff
@@ -68,6 +68,11 @@ func (c gitClient) download(ctx context.Context, dst string, opt cloneOptions) e
 }
 
 func (c gitClient) latestCommitID(ctx context.Context, opt fetchOptions) (string, error) {
+	if opt.referenceName == "" {
+		// allow the remote git to give us whatever is default.
+		return "", nil
+	}
```

Comment:
That would mean that by default a recorded commit id will be `""` and it would be the same every time Portainer would try to fetch a latest commit id. That would result in a repo not being pulled and redeployed, as the commit would be always the same - `""`. I'm afraid there is a bug here.
We may need to figure out how to get a last commit id for the default branch.

Reply:
yeah, damn. in the testing i did, i failed to get enough info before doing the download - sounds like someone's going to need to do another day or two :/